src/graphcast_agent/waku_handling.rs

```diff
@@ -496,25 +488,35 @@ pub fn peers_data(
 
 /// Check for peer connectivity, try to reconnect if there are disconnected peers
 pub fn network_check(node_handle: &WakuNodeHandle<Running>) -> Result<(), WakuHandlingError> {
-    peers_data(node_handle)?
-        .iter()
-        // Get unconnected peers and try to reconnect
-        .filter(|&peer| !peer.connected())
-        .map(|peer: &WakuPeerData| {
-            debug!(
-                peer = tracing::field::debug(&peer),
-                "Disconnected peer data"
-            );
-            node_handle.connect_peer_with_id(peer.peer_id(), None)
-        })
-        .for_each(|res| {
-            if let Err(e) = res {
-                debug!(
-                    error = tracing::field::debug(&e),
-                    "Could not connect to peer"
-                );
+    let peers = peers_data(node_handle)?;
+
+    for peer in peers.iter() {
+        let contains_filter = peer
+            .protocols()
+            .iter()
+            .any(|p| p == "/vac/waku/filter/2.0.0-beta1");
+        let contains_lightpush = peer
+            .protocols()
+            .iter()
+            .any(|p| p == "/vac/waku/lightpush/2.0.0-beta1");
+        let contains_relay = peer
+            .protocols()
+            .iter()
+            .any(|p| p == "/vac/waku/relay/2.0.0");
+
+        if contains_filter && contains_lightpush && contains_relay {
```

Comment:
this might not exactly match the requirement for what the local node wants. For instance we typically expect relay to be off locally, so this would be restricting most of the peers. Can we condition with `WakuNodeConfig`'s fields like `relay` and `filter`?